src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +25,122 @@ interface GetCourseDetailsResponse {
-   */
-  getStudentsToHide(courseId: string): string[] {
-    return this.studentDetailsList
-        .filter((student: StudentDetails) => student.courseId === courseId && !student.isChecked)
-        .map((student: StudentDetails) => student.email);
-  }
+   * Loads students of a specified course.
+   */
+  loadStudents(courseId: string): void {
+    this.httpRequestService.get('/students', {
+      courseid: courseId,
+    }).subscribe((students: Students) => {
+
+      const sections: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
+        const term: string = x.sectionName;
+        (acc[term] = acc[term] || []).push(x);
+        return acc;
+      }, {});
+
+      const teams: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
+        const term: string = x.teamName;
+        (acc[term] = acc[term] || []).push(x);
+        return acc;
+      }, {});
+
+      this.courseStats[courseId] = {
+        students: students.students.length,
+        sections: Object.keys(sections).length,
+        teams: Object.keys(teams).length,
+      };
+
+      Object.keys(sections).forEach((key: string) => {
+        const studentsInSection: Student[] = sections[key];
+
+        const data: StudentListStudentData[] = [];
+        studentsInSection.forEach((student: Student) => {
+          const studentData: StudentListStudentData = {
+            name : student.name,
+            status : (student.joinState === JoinState.JOINED) ? 'Joined' : 'Yet to Join',
+            email : student.email,
+            team : student.teamName,
+          };
+          data.push(studentData);
+        });
 
-  /*------------------------------Functions to formulate data for student list from item------------------------*/
-  /**
-   * Formulate data for student list.
-   */
-  getStudentListSectionDataForCourse(courseId: string): StudentListSectionData[] {
-    return this.courseStudentListSectionDataMap[courseId] || [];
+        this.loadPrivilege(courseId, key, data);
+      });
+    }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
   }
 
   /**
-   * Formulate data for student list.
+   * Loads privilege of an instructor for a specified course and section.
    */
-  getStudentListSectionDataFromCourseDetails(courseDetails: CourseDetails): StudentListSectionData[] {
-    const sections: StudentListSectionData[] = [];
-    courseDetails.sections.forEach((section: SectionDetails) => {
-      const sectionData: StudentListSectionData = this.mapSectionForStudentList(section);
-      sections.push(sectionData);
-    });
+  loadPrivilege(courseId: string, sectionName: string, students: StudentListStudentData[]): void {
+    this.httpRequestService.get('/instructor/privilege', {
+      courseid: courseId,
+      sectionname: sectionName,
+    }).subscribe((instructorPrivilege: InstructorPrivilege) => {
+      const temp: StudentListSectionData = {
```

Comment:
Avoid just naming it `temp`? It could make others confused

Reply:
Yes updated!